lib/chef/provider/systemd_unit.rb

```diff
@@ -0,0 +1,172 @@
+#
+# Author:: Nathan Williams (<nath.e.will@gmail.com>)
+# Copyright:: Copyright 2016, Nathan Williams
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/provider"
+require "chef/mixin/which"
+require "chef/mixin/shell_out"
+
+class Chef
+  class Provider
+    class SystemdUnit < Chef::Provider
+      include Chef::Mixin::Which
+
+      provides :systemd_unit, os: "linux"
+
+      def load_current_resource
+        @current_resource = Chef::Resource::SystemdUnit.new(new_resource.name)
+
+        current_resource.user(new_resource.user)
+        current_resource.enabled(enabled?)
+        current_resource.active(active?)
+        current_resource.masked(masked?)
+        current_resource.static(static?)
+
+        current_resource
+      end
+
+      def action_enable
+        if current_resource.static
+          Chef::Log.warn("#{new_resource.name} is a static unit, enabling is a NOP.")
+        end
```

Comment:
is it really necessary to warn the user here?  if a NOP and its always a NOP does anyone care that much?  its vaguely 'unclean' code but it feels like we can safely make the user's life easier here by not bugging them about something where the action we can take is obvious...